sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosContainer.java

```diff
@@ -218,8 +218,8 @@ CosmosItemResponse mapItemResponseAndBlock(Mono<CosmosAsyncItemResponse> itemMon
      * @param options the options
      * @return the iterator
      */
-    public Iterator<FeedResponse<CosmosItemProperties>> queryItems(String query, FeedOptions options) {
-        return getFeedIterator(this.containerWrapper.queryItems(query, options));
+    public <T> Iterator<FeedResponse<T>> queryItems(String query, FeedOptions options, Class<T> klass) {
```

Comment:
Should we have an overload that defaults the class type argument to CosmosItemProperties.class?

Reply:
Bhaskar do we expect majority of use cases to have CosmosItemProperties in query? if so maybe it is worth considering an overload as Milis suggested.

Reply:
I don't see too many use cases for this. We have most of our tests using this and hence kept the CosmosItemProperties special type for the time being.  We can discuss about this if we want to support it as a type